common/network-common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -117,19 +118,26 @@ public static String genStreamChunkId(long streamId, int chunkId) {
 
   @Override
   public void connectionTerminated(Channel channel) {
+    LinkedList<StreamState> removedStates = new LinkedList<>();
     // Close all streams which have been associated with the channel.
     for (Map.Entry<Long, StreamState> entry: streams.entrySet()) {
       StreamState state = entry.getValue();
       if (state.associatedChannel == channel) {
-        streams.remove(entry.getKey());
+        removedStates.add(streams.remove(entry.getKey()));
+      }
+    }
 
-        // Release all remaining buffers.
+    for (StreamState state: removedStates) {
+      // Release all remaining buffers.
+      try {
         while (state.buffers.hasNext()) {
           ManagedBuffer buffer = state.buffers.next();
```

Comment:
Once we got RuntimeException, don't we just fail? Is there memory leak?

Reply:
https://github.com/apache/spark/blob/1b0570c6af25849d46ddd2d108d2fa949616b756/common/network-common/src/main/java/org/apache/spark/network/server/TransportRequestHandler.java#L87-L97

`TransportRequestHandler.channelInactive` will log error message and swallow the exception, so no, we don't fail because of exception thrown here.

Reply:
oh I see. Then this change may also swallow the RuntimeException so `channelInactive` cannot log as before? Should we rethrow the same exception?

Reply:
So I think there're two options here with achieving goal;

1. store any exception between exceptions (as we'll try to release from all the streams) and rethrow, then channelInactive will log it. other exceptions should be logged as well but maybe at here (or a new exception containing all exceptions and throw it instead).

2. catch them, and log and swallow at here.

Technically the change from logging side would be logger name which may not a big deal, but it's also valid concern if we think channelInactive is the one to decide how to handle the exception.

Reply:
yea, I think it might make the caller (channelInactive or others if any) thinks everything is fine inside `connectionTerminated`. Maybe not big deal for now, but sounds a potential concern.

We might have multiple exception during releasing buffers. We can rethrow a `RuntimeException` if any exception happens during releasing buffers.

Reply:
OK thanks for the input. @hensg Could you follow up the comment above? Thanks!